eng/scripts/SignAndHash-MavenPackages.ps1

```diff
@@ -0,0 +1,177 @@
+    ($_ -ne $pomArtifact) -and
+    ($_ -ne $fileArtifact) -and
+    ($_ -ne $javadocArtifact) -and
+    ($_ -ne $sourcesArtifact)
+  }
+
+  return [ordered]@{
+    FullyQualifiedName= $packageDetail.FullyQualifiedName;
+    GroupId= $packageDetail.GroupId;
+    ArtifactId= $packageDetail.ArtifactId;
+    Version= $packageDetail.Version;
+    PomArtifact= $pomArtifact;
+    FileArtifact= $fileArtifact;
+    JavadocArtifact= $javadocArtifact;
+    SourcesArtifact= $sourcesArtifact;
+    AdditionalArtifacts= $additionalArtifacts;
+  }
+}
+
+
+Write-Host "PS Script Root is: $PSScriptRoot"
+Write-Host "Path is: $Path"
+Write-Host "DestinationPath is: $DestinationPath"
+Write-Host "GPG Executable Path is: $GPGExecutablePath"
+Write-Host "Group ID Filter is: $GroupIDFilter"
+Write-Host "Artifact ID Filter is: $ArtifactIDFilter"
+
+Write-Host "Getting filtered package details."
+
+[array]$packageDetails = Get-FilteredMavenPackageDetails -ArtifactDirectory $Path -GroupIDFilter $GroupIDFilter -ArtifactIDFilter $ArtifactIDFilter
+| ForEach-Object { ConvertTo-DeploymentDetails $_ }
+
+Write-Host "Found $($packageDetails.Length) packages to publish:"
+$packageDetails | ForEach-Object { Write-Host $_.FullyQualifiedName }
+
+if ($packageDetails.Length -eq 0) {
+  throw "Aborting, no packages to publish."
+}
+
+Write-Host "Creating destination directory $DestinationPath"
+if (Test-Path $DestinationPath) {
+  Remove-Item $DestinationPath -Force -Recurse | Out-Null
+}
+New-Item -Path $DestinationPath -Type Directory | Out-Null
+
+$DestinationPath = Resolve-Path -Path $DestinationPath
+$destinationPathUri = $([Uri]$DestinationPath).AbsoluteUri
+
+foreach ($packageDetail in $packageDetails) {
+  Write-Host "GPG signing and publishing package: $($packageDetail.FullyQualifiedName)"
+
+  $pomOption = "-DpomFile=$($packageDetail.PomArtifact.File.FullName)"
+  Write-Host "POM Option is: $pomOption"
+
+  $fileOption = "-Dfile=$($packageDetail.FileArtifact.File.FullName)"
+  Write-Host "File Option is: $fileOption"
+
+  $javadocOption = ""
+  if (-not $packageDetail.JavadocArtifact) {
+    Write-Host "No JavaDoc artifact, omitting JavaDoc Option"
+  } else {
+    $javadocOption = "-Djavadoc=$($packageDetail.JavadocArtifact.File.FullName)"
+    Write-Host "JavaDoc Option is: $javadocOption"
+  }
+
+  $sourcesOption = ""
+  if (-not $packageDetail.SourcesArtifact) {
+    Write-Host "No Sources artifact, omitting Sources Option"
+  } else {
+    $sourcesOption = "-Dsources=$($packageDetail.SourcesArtifact.File.FullName)"
+    Write-Host "Sources Option is: $sourcesOption"
+  }
+
+  foreach ($additionalArtifact in $packageDetail.AdditionalArtifacts) {
+    Write-Host "Additional associated artifact is: $($additionalArtifact.File.FullName)"
+  }
+
+  $commaDelimitedFileNames = ""
+  $commaDelimitedClassifiers = ""
+  $commaDelimitedTypes = ""
+
+  if ($null -ne $packageDetail.AdditionalArtifacts) {
+    foreach($additionalArtifact in $packageDetail.AdditionalArtifacts) {
```

Comment:
Nit: Probably simpler here to use an array and then use `-join` rather than constructing the string and then cutting the leading comma off.

Reply:
Agreed.   Same as before though.   I'd rather address in a refactor pr than add a different implementation in this file.

https://github.com/Azure/azure-sdk-for-java/blob/d55e2cfcf4ed87a1ad0d083baa1cb339ae771d00/eng/scripts/Publish-MavenPackages.ps1#L110-L122